bats/tests/k8s/helm-install-rancher.bats

```diff
@@ -13,12 +13,95 @@ local_setup() {
     helm repo update
 }
 
+# Get the minimum Kubernetes version the given rancher-latest/rancher helm chart
+# does _not_ support; i.e. the chart supports < [output].
+get_chart_unsupported_version() {
+    output="" # Hack: tell shellcheck about this variable
+    local chart_version=$1
+    run helm show chart rancher-latest/rancher --version "$chart_version"
+    assert_success || return
+    run awk '/^kubeVersion:/ { $1 = ""; print }' <<<"$output"
+    assert_success || return
+    # We only support kubeVersion of form "< x.y.z"
+    assert_output --regexp '^[[:space:]]*<[[:space:]]*[^[:space:]]+$' || return
+    awk '{ print $2 }' <<<"$output"
+}
+
+# Try to determine the best chart version for the current Kubernetes version
+# (as specified in $RD_KUBERNETES_PREV_VERSION).  Saves the resulting version
+# in rancher_chart_version (and should be restored via load_var).
+determine_chart_version() {
+    if [[ -n $RD_RANCHER_IMAGE_TAG ]]; then
+        # If a version is given, check that it's compatible.
+        assert_rancher_chart_compatible "${RD_RANCHER_IMAGE_TAG#v}" || return
+        rancher_chart_version=${RD_RANCHER_IMAGE_TAG#v}
```

Comment:
Shouldn't we use `local` here? And we can avoid duplicating the substitution:

```suggestion
        local rancher_chart_version=${RD_RANCHER_IMAGE_TAG#v}
        assert_rancher_chart_compatible "$rancher_chart_version" || return
```

But more generally, I feel that an `assert_*` function should not call `skip*`; it should simply assert a condition, and fail if it isn't true. Any additional action should not be hidden inside the `assert`. So I would prefer something like:

```suggestion
        local rancher_chart_version=${RD_RANCHER_IMAGE_TAG#v}
        if ! rancher_chart_compatible "$rancher_chart_version"; then
            skip_k3s_version "Rancher Manager ${rancher_chart_version} is incompatible with Kubernetes ${RD_KUBERNETES_PREV_VERSION}"
        fi
```